spike_main/spike.cc

```diff
@@ -176,20 +177,32 @@ static std::vector<mem_cfg_t> parse_mem_layout(const char* arg)
       size += PGSIZE - size % PGSIZE;
 
     if (size != size0) {
-      fprintf(stderr, "Warning: the memory at  [0x%llX, 0x%llX] has been realigned\n"
-                      "to the %ld KiB page size: [0x%llX, 0x%llX]\n",
+      fprintf(stderr, "Warning: the memory at "
+                      "[0x%" PRIx64 ", 0x%" PRIx64 "] has been realigned\n"
+                      "to the %ld KiB page size: "
+                      "[0x%" PRIx64 ", 0x%" PRIx64 "]\n",
```

Comment:
These changes are beneficial but independent of the new functionality you've added, so should be in its own commit.

Reply:
ok, will move this part to a separate PR then

Reply:
Not a separate PR. It's related to the other change, so it's reasonable to include here in this PR.

Reply:
The patch was refactored (a little bit)